src/components/FeatureSetsPanel/featureSetPanel.util.js

```diff
@@ -1,9 +1,16 @@
+import { isNil } from 'lodash'
+
 export const nameValidationPattern = /^(?=[\S\s]{1,56}$)[a-z0-9]([-a-z0-9]*[a-z0-9])?$/
 
 export const checkValidation = (newFeatureSet, setValidation, validation) => {
   const externalOfflineTarget = newFeatureSet.spec.targets.find(
     targetKind => targetKind.name === 'externalOffline'
   )
+  const isPartitionByTimeExist = Boolean(
+    newFeatureSet.spec.targets.find(
+      target => !isNil(target.time_partitioning_granularity)
+    )
+  )
```

Comment:
Instead of using `find` and wrap its result with `Boolean(…)` you can use `some`.

| | `Array.prototype.find`  | `Array.prototype.some` |
| ------------- | ------------- | ------------- |
| Matches | The first item that matches the provided criterion | The first item that matches the provided criterion |
| Returns when match found | The array item | `true` |
| Returns when match **not** found | `undefined` | `false` |

Moreover, you can change `isNil` (which matched `undefined` and `null`) to `isEmpty` (which also matches an empty string:

```suggestion
  const isPartitionByTimeExist = newFeatureSet.spec.targets.some(
    target => !isEmpty(target.time_partitioning_granularity)
  )
```

Honestly, I'm not sure we need `isNil` or `isEmpty` here.
`Boolean` will also give `true` for a non-empty string, and `false` to `undefined`, `null` and `''`.
```suggestion
  const isPartitionByTimeExist = newFeatureSet.spec.targets.some(
    target => Boolean(target.time_partitioning_granularity)
  )
```